### ISSUE TITLE
Scripts are not loading as expected

### ISSUE THREAD

**Issue by anuj-scanova** [scullyio/scully#1609]
### Description
Pagespeed score is very low in comparison with SSR

#### 🔬 Minimal Reproduction
Add the Google tag manager script to the footer and generate a build
```html
<script>
  function loadGtm() {
    setTimeout(() => {
      (function (w, d, s, l, i) {
        w[l] = w[l] || [];
        w[l].push({
          'gtm.start':
            new Date().getTime(), event: 'gtm.js'
        });
        var f = d.getElementsByTagName(s)[0],
          j = d.createElement(s), dl = l != 'dataLayer' ? '&l=' + l : '';
        j.async = true;
        j.src =
          'https://www.googletagmanager.com/gtm.js?id=' + i + dl;
        f.parentNode.insertBefore(j, f);
      })
      (window, document, 'script', 'dataLayer', 'GTM-XXXXXXX');
    }, 4000);
  }

  window.onload = loadGtm;
</script>
```

The script should load after 4 seconds. This is working well with Angular SSR and the external scripts are not detected in the Google page speed. While the Scully build of the same application, Google page speed is detecting all external scripts.

<img width="1056" alt="Screen Shot 2022-09-01 at 12 40 21 PM" src="https://user-images.githubusercontent.com/105233146/187853714-fff06930-b32a-4b34-acee-80a15500b29e.png">

#### 💻Your Environment

**Angular Version:**

<pre><code>

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 14.1.3
Node: 16.16.0
Package Manager: npm 8.11.0 
OS: darwin arm64

Angular: 14.1.3
... animations, cli, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic
... platform-server, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1401.3
@angular-devkit/build-angular   14.1.3
@angular-devkit/core            14.1.3
@angular-devkit/schematics      14.1.3
@nguniversal/builders           14.1.0
@nguniversal/express-engine     14.1.0
@schematics/angular             14.1.3
rxjs                            7.5.6
typescript                      4.7.4

</code></pre>

**Scully Version:**

<pre><code>

"dependencies": {
    "@angular/animations": "^14.1.3",
    "@angular/common": "^14.1.3",
    "@angular/compiler": "^14.1.3",
    "@angular/core": "^14.1.3",
    "@angular/forms": "^14.1.3",
    "@angular/platform-browser": "^14.1.3",
    "@angular/platform-browser-dynamic": "^14.1.3",
    "@angular/platform-server": "^14.1.3",
    "@angular/router": "^14.1.3",
    "@ng-bootstrap/ng-bootstrap": "^13.0.0",
    "@nguniversal/express-engine": "^14.1.0",
    "@popperjs/core": "^2.11.5",
    "@scullyio/init": "^2.1.36",
    "@scullyio/ng-lib": "^2.1.0",
    "@scullyio/scully": "^2.1.0",
    "@scullyio/scully-plugin-puppeteer": "^2.1.0",
    "bootstrap": "^5.2.0",
    "express": "^4.15.2",
    "jquery": "^3.6.0",
    "ng-recaptcha": "^10.0.0",
    "rxjs": "~7.5.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^14.1.3",
    "@angular/cli": "~14.1.3",
    "@angular/compiler-cli": "^14.1.3",
    "@angular/localize": "^14.1.3",
    "@nguniversal/builders": "^14.1.0",
    "@scullyio/scully-plugin-critical-css": "^2.1.36",
    "@types/express": "^4.17.0",
    "@types/jasmine": "~4.0.0",
    "@types/node": "^14.15.0",
    "extract-css": "^3.0.0",
    "jasmine-core": "~4.1.0",
    "karma": "~6.3.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.0.0",
    "karma-jasmine-html-reporter": "~1.7.0",
    "purgecss": "^4.1.3",
    "typescript": "~4.7.2"
  },
</code></pre>

## 🔥 Exception or Error

<pre><code>

</code></pre>

**Comment by SanderElias** [comment]
@anuj-scanova That is because those scripts are never run in Universal, but they _are_ when using Scully.
Make sure you are not executing those things when you are rendering through Scully.
If you want to do that in native JS, take a look [here[(https://github.com/scullyio/scully/blob/main/libs/ng-lib/src/lib/utils/isScully.ts#L2-L4)

TLDR: wrap your JS code you don't want to run when rendering like this:
```js
if (window['ScullyIO'] !== 'running') {
   /** insert your code here */
 }
```

**Comment by anuj-scanova** [comment]
@SanderElias Thanks for the response but adding the script to `if` condition is now no longer loading the script.
> That is because those scripts are never run in Universal

I have confirmed that the scripts are loading in the Universal as well with the `timeout` script and due to timeout, it is not reporting in Google page speed.
While in the case of Scully, the scripts are detected in Google page speed, which looks like timeout is not working with scully.

**Comment by SanderElias** [comment]
Can you explain that a bit more?
You mean when you run this in Universal, the server does wait 4 seconds,
runs the script, before shipping the code to the client?  That would entail
a 4 seconds delay in delivering every page right? I'm fairly sure that
would not do well in PageInsigths.

Scully doesn't have _any_ influence on how a script is being executed in
the browser. The workaround I gave you above will make sure the script
isn't run during rendering, but it will run on a normal page load.  Perhaps
you need a bigger delay as 4 seconds in Scully, because it will deliver
stuff faster, and it executes your script within the cutoff time of
PageInsights, where your SSR version doesn't.

Op ma 5 sep. 2022 om 10:18 schreef Anuj Sharma ***@***.***>:

**Comment by anuj-scanova** [comment]
```javascript
<script>
  function loadGtm() {
    setTimeout(() => {
      (function (w, d, s, l, i) {
        w[l] = w[l] || [];
        w[l].push({
          'gtm.start':
            new Date().getTime(), event: 'gtm.js'
        });
        var f = d.getElementsByTagName(s)[0],
          j = d.createElement(s), dl = l != 'dataLayer' ? '&l=' + l : '';
        j.async = true;
        j.src =
          'https://www.googletagmanager.com/gtm.js?id=' + i + dl;
        f.parentNode.insertBefore(j, f);
      })
      (window, document, 'script', 'dataLayer', 'GTM-XXXXXXX');
    }, 4000);
  }

  window.onload = loadGtm;
</script>
```
This code snippet is added to the `index.html` file and I want to load the GTM script after 4000ms after the window has loaded.

In the universal case, the index.html file is returned with the script content and loads after 4000ms of window load on the client side (not on the server side).

Same behavior I want from the Scully as well, as this script is not part of any component or routing.

I just checked the compiled source code and found that the script is executed while processing of `npm run scully -- --scanRoutes` and the following script is added to the head of the generated HTML file, which it was expected to be added on client-side after the window is loaded.
```
<script async src="https://www.googletagmanager.com/gtm.js?id=GTM-WV9J9TH"></script>
```

How can I bypass execution of the scripts while generating static site?
